torchrl/data/replay_buffers/storages.py

```diff
@@ -550,6 +550,9 @@ def shape(self):
         if _total_shape is not None:
             return torch.Size([self._len_along_dim0] + list(_total_shape[1:]))
 
+    # TODO: Without this disable, compiler recompiles for back-to-back calls.
+    # Figuring out a way to avoid this disable would give better performance.
+    @torch._dynamo.disable()
     def _rand_given_ndim(self, batch_size):
```

Comment:
I haven't been able to figure this one out yet, unfortunately. Even if remove the `len(storage)` and use `self._len_value` directly, it still puts a guard around `_len_value`. I even tried completely removing any code that seems special about `_len_value` (the property decorator and anything related to `mp.Value`), and it still puts a guard around it.

I can't seem to find anything in the output from TORCH_TRACE that tells exactly why there's a guard on it. I've attached a trace of running the benchmark after commenting out the `@torch._dynamo.disable()`.

[tl_out.zip](https://github.com/user-attachments/files/17582288/tl_out.zip)

Reply:
that's ok, let's deactivate like you do here.
Perhaps we could isolate this and put a test around that graph break to make sure that if the compiler ends up hadling this properly we're aware of it and we can remove the disable() decorator?

Reply:
Good idea, I've added a test for that